src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -404,6 +394,25 @@ internal Symbol BindNamespaceOrTypeOrAliasSymbol(ExpressionSyntax syntax, Diagno
             }
         }
 
+        private Symbol BindTupleExpression(TupleTypeSyntax syntax, DiagnosticBag diagnostics)
+        {
+            var count = syntax.Elements.Count;
+            var symbols = ArrayBuilder<TypeSymbol>.GetInstance(count);
+            var names = ArrayBuilder<string>.GetInstance(count);
+
+            for (int i = 0; i < count; i++)
+            {
+                var argumentSyntax = syntax.Elements[i];
+                var argumentSymbol = BindTypeArgument(argumentSyntax.Type, diagnostics);
+                symbols.Add(argumentSymbol);
+
+                var name = argumentSyntax.Name?.Identifier.ValueText;
```

Comment:
For error recovery we may need to handle cases like   `(int foo, int bar, int ` - i.e with some names missing. Forcing them to be "ItemX"  might be sufficient.